lms/djangoapps/courseware/access.py

```diff
@@ -323,6 +324,14 @@ def can_load():
         if descriptor.visible_to_staff_only and not _has_staff_access_to_descriptor(user, descriptor, course_key):
             return False
 
+        # if course has pre-requisite course not passed by user
+        if settings.FEATURES['ENABLE_PREREQUISITE_COURSES'] \
+                and not _has_staff_access_to_descriptor(user, descriptor, course_key) \
+                and hasattr(descriptor, 'pre_requisite_courses') and descriptor.pre_requisite_courses \
+                and not user.is_anonymous() \
```

Comment:
we are not allowing anonymous access we are don't want to pull fulfillments of anonymous user since we don't have `id` attribute of user.